c++/src/kj/async-prelude.h

```diff
@@ -33,12 +33,12 @@
 // TODO(someday): Support coroutines with -fno-exceptions.
 #if !KJ_NO_EXCEPTIONS
 #ifdef __has_include
-#if __cpp_coroutines && __has_include(<coroutine>)
+#if __cpp_coroutines && (__cpp_impl_coroutine >= 201902L) && __has_include(<coroutine>)
```

Comment:
Looking at https://en.cppreference.com/w/cpp/feature_test, I don't actually see `__cpp_coroutines` at all. I wonder if that feature macro is technically defined just for the Coroutines TS, and we should only be relying on `__cpp_impl_coroutine` here to detect C++20 coroutines? E.g., I think this would work:

```cpp
#if (__cpp_impl_coroutine >= 201902L) && __has_include(<coroutine>)
// ...
#elif (__cpp_coroutines >= 201703L) && __has_include(<experimental/coroutine>)
// ...
#endif
```

Does that make sense to you?

Reply:
Will check

Reply:
Your fix works fine as well! I've updated the PR